server/src/users/users.service.ts

```diff
@@ -1,15 +1,19 @@
-import { Injectable } from '@nestjs/common';
+import { HttpException, HttpStatus, Injectable } from '@nestjs/common';
 import { User } from './entities/user.entity';
 import { Repository } from 'typeorm';
 import { CreateUserDto } from './dto/create-user.dto';
 import { UpdateUserDto } from './dto/update-user.dto';
 import { InjectRepository } from '@nestjs/typeorm';
+import * as bcrypt from 'bcrypt';
 
 @Injectable()
 export class UsersService {
     constructor(@InjectRepository(User) private usersRepository: Repository<User>) {}
-    create(createUserDto: CreateUserDto) {
-        return this.usersRepository.save(createUserDto)
+
+    async create(createUserDto: CreateUserDto) {
+        const hashedPw = await bcrypt.hash(createUserDto.password, parseInt(process.env.BCRYPT_WORK_FACTOR));
+        createUserDto.password = hashedPw;
+        return this.usersRepository.save(createUserDto);
```

Comment:
```
jamesrogers@jamesrogpremiseMBP-2 web-app % curl --request POST \
  --url http://localhost:3001/api/users \
  --header 'Content-Type: application/json' \
  --data '{
        "first_name": "TestFirst",
        "last_name": "TestLast",
        "email": "test@test.com",
        "password": "testpassword"
        }'
{"first_name":"TestFirst","last_name":"TestLast","email":"test@test.com","password":"$2b$10$ZFlYKLACsaNmeUFbOxhzr.NAgBl1G3BsnheWYkRFQTVgTzQFtHlDm","id":1}%                                                                                                                                     
jamesrogers@jamesrogpremiseMBP-2 web-app % 
```

create user post request is returning password with it. might wrap this return val (and i suspect the other return vals in this file) with a `getUserSansPw`

```
const getUserSansPw = (user: CreateUserDto): Omit< CreateUserDto, 'password'> => {
  const { password, ...result } = user;
  return result;
};
```
(https://www.typescriptlang.org/docs/handbook/utility-types.html#omittype-keys)

Reply:
Oo Omit, that is very handy. I am planning to have a PR that has the user routes have more informative responses. I could address this issue then.

Reply:
That works too!